fog/view/enclave/impl/src/lib.rs

```diff
@@ -61,6 +61,63 @@ where
             logger,
         }
     }
+
+    fn query_impl(
+        &self,
+        plaintext_request: &[u8],
+        untrusted_query_response: UntrustedQueryResponse,
+    ) -> Result<Vec<u8>> {
+        let req: QueryRequest = mc_util_serial::decode(plaintext_request).map_err(|e| {
+            log::error!(self.logger, "Could not decode user request: {}", e);
+            Error::ProstDecode
```

Comment:
:wrench: It looks like this was `ProstDecode` previously, but looking at `mc_util_serial::decode()` it looks like this is a serde decode
```suggestion
            Error::SerdeDecode
```

Reply:
We shouldn't have any serde in encrypted messages.

Reply:
nvm, my bad
```rust
pub use prost::{DecodeError, EncodeError, Message};
...
pub fn decode<T: Message + Default>(buf: &[u8]) -> Result<T, DecodeError> {
    T::decode(buf)
}
```
That looks like prost.
I was looking at the `pub mod decode`